application/common/components/Emailer.php

```diff
@@ -285,6 +340,134 @@ public function shouldSendWelcomeMessageTo($user, $changedAttributes)
             && array_key_exists('current_password_id', $changedAttributes)
             && is_null($changedAttributes['current_password_id']);
     }
+
+    /**
+     * @param User $user The User in question.
+     * @return bool
+     */
+    public function shouldSendGetBackupCodesMessageTo($user)
```

Comment:
We may need to add something like...
```
...
&& !$user->hasReceivedMessage(EmailLog::MESSAGE_TYPE_GET_BACKUP_CODES)
```
... to each of these new `shouldSend...()` methods so that we don't send these emails every night (or whatever cron schedule we have these on) as soon as the user matches the other criteria.

We may actually need to add a `hasReceivedMessageRecently(string $messageType, int $recentDays)` method so that we can send these reminder emails once every X days.

Reply:
Done